src/routes/_surfaces/workspace/explore/leaderboards/LeaderboardDisplay.svelte

```diff
@@ -1,105 +1,109 @@
     ShortHandSymbols,
   } from "$lib/util/humanize-numbers";
+  import { useQueryClient } from "@sveltestack/svelte-query";
   import { onDestroy, onMount } from "svelte";
-  import type { Readable } from "svelte/store";
   import Leaderboard from "./Leaderboard.svelte";
 
   export let metricsDefId: string;
   export let whichReferenceValue: string;
 
-  let metricsExplorer: Readable<MetricsExplorerEntity>;
-  $: metricsExplorer = getMetricsExplorerById(metricsDefId);
+  const queryClient = useQueryClient();
+
+  let metricsExplorer: MetricsExplorerEntity;
+  $: metricsExplorer = $metricsExplorerStore.entities[metricsDefId];
 
-  let dimensions: Readable<DimensionDefinitionEntity[]>;
-  $: dimensions = getDimensionsByMetricsId(metricsDefId);
+  // query the `/meta` endpoint to get the metric's measures and dimensions
+  $: metaQuery = useGetMetricViewMeta(metricsDefId);
+  $: dimensions = $metaQuery.data.dimensions;
+  $: measures = $metaQuery.data.measures;
 
-  let measureField: Readable<string>;
-  $: if ($metricsExplorer?.leaderboardMeasureId)
-    measureField = getMeasureFieldNameByIdAndIndex(
-      $metricsExplorer.leaderboardMeasureId,
-      $metricsExplorer.measureIds.indexOf(
-        $metricsExplorer?.leaderboardMeasureId
-      )
+  $: activeMeasure =
+    measures &&
+    measures.find(
+      (measure) => measure.id === metricsExplorer?.leaderboardMeasureId
     );
 
-  $: measures = getMeasuresByMetricsId(metricsDefId);
-  $: leaderboardMeasureDefinition = $measures.find(
-    (measure) => measure.id === $metricsExplorer?.leaderboardMeasureId
-  );
-  // get the expression so we can determine if the measure is summable
-  $: expression = leaderboardMeasureDefinition?.expression;
   $: formatPreset =
-    leaderboardMeasureDefinition?.formatPreset ?? NicelyFormattedTypes.HUMANIZE;
+    activeMeasure?.formatPreset ?? NicelyFormattedTypes.HUMANIZE;
 
-  let bigNumberEntity: Readable<BigNumberEntity>;
-  $: bigNumberEntity = getBigNumberById(metricsDefId);
   let referenceValue: number;
 
-  $: if ($bigNumberEntity && $measureField) {
+  $: totalsQuery = useGetMetricViewTotals(
+    metricsDefId,
+    {
+      measures: metricsExplorer?.selectedMeasureIds,
+      filter: metricsExplorer?.filters,
+      time: {
+        start: metricsExplorer?.selectedTimeRange?.start,
+        end: metricsExplorer?.selectedTimeRange?.end,
+      },
+    },
+    false,
+    {
+      enabled: $metaQuery?.isFetched,
+    }
+  );
+  // TODO: find a way to have a single request when there are no filters
+  const referenceValueQuery = useGetMetricViewTotals(
+    metricsDefId,
+    {
+      measures: metricsExplorer?.selectedMeasureIds,
+      filter: undefined,
+      time: {
+        start: metricsExplorer?.selectedTimeRange?.start,
+        end: metricsExplorer?.selectedTimeRange?.end,
+      },
+    },
+    true,
+    {
+      enabled: $metaQuery?.isFetched,
+    }
+  );
+
+  $: if ($totalsQuery && $referenceValueQuery && activeMeasure?.sqlName) {
```

Comment:
This doesn't seem to work since the leaderboard values aren't updating

Reply:
This statement should probably be re-written. In each case, you can check something like this:

```javascript
    if (activeMeasureName in $totalsQuery.data.data) {
      referenceValue = $totalsQuery.data.data[activeMeasureName];
    }
```

And it should work.

Reply:
`in` check is not needed, just checking for existence of response is enough.